src/domain/table-columns/repositories/TableColumnsRepository.ts

```diff
@@ -0,0 +1,11 @@
+import { ConfigRepository } from "../../config/repositories/ConfigRepository";
+import { TableColumn } from "../entities/TableColumn";
+
+export interface TableColumnsRepositoryConstructor {
+    new (configRepository: ConfigRepository): TableColumnsRepository;
+}
+
+export interface TableColumnsRepository {
+    getColumns(id: string): Promise<TableColumn[]>;
+    saveColumns(id: string, columns: string[]): Promise<void>;
+}
```

Comment:
No need to change since it was my proposal, just a note for the future:

I forgot to remove the name of the entity in the methods:

getColumns -> get
saveColumns -> save

because the repository's name `TableColumnsRepository` already implies that the operations are related to `TableColumns` the method names can be shortened to get and save without losing clarity.

Another example in the [skeleton app](https://github.com/EyeSeeTea/dhis2-app-skeleton/blob/master/src/domain/repositories/UserRepository.ts#L5) `getCurrent` instead `getCurrentUser`